src/main/java/net/neoforged/neoforge/transfer/transaction/snapshots/NotifyingSnapshotJournal.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer.transaction.snapshots;
+
+import net.neoforged.neoforge.transfer.transaction.TransactionContext;
+import org.jetbrains.annotations.Nullable;
+
+// TODO notes for reviewers: This is used in at least 4 different places in Neoforge implementations later, and use roughly the same layout.
```

Comment:
Remove it. The other slices have not been reviewed yet so don't use them to justify this existing. If we need it later it can be readded later.